### PR TITLE
Feature/ Helia React Vite Example

### PR DESCRIPTION
**Pull request by polus-arcticus**
Thought I'd try writing an example for Vite+React

### CODE REVIEW

File: examples/helia-vite-react/src/hooks/useCommitText.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        setCid(cid)
+        setCidString(cid.toString())
+        console.log('Added file:', cid.toString())
+      } catch (e) {
+        console.error(e)
+      }
+    } else {
+      console.log('please wait for helia to start')
+    }
+  }, [error, starting, helia])
```
Comment (SgtPooki) [inline examples/helia-vite-react/src/hooks/useCommitText.jsx:30]:
```suggestion
  }, [error, starting, helia, fs])
```

**Comment by polus-arcticus** [comment]
Thanks for the comments @SgtPooki :tada:  I'll make these changes

File: examples/helia-vite-react/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+</p>
+
+<h3 align="center"><b>Helia with React+Vite</b></h3>
+
+<p align="center">
+  <img src="https://raw.githubusercontent.com/jlord/forkngo/gh-pages/badges/cobalt.png" width="200">
+  <br>
+  <a href="https://ipfs.github.io/helia/modules/helia.html">Explore the docs</a>
+  ·
+  <a href="https://codesandbox.io/">View Demo</a>
```
[the remark this answers is not in the document]
Comment (polus-arcticus) [reply examples/helia-vite-react/README.md]: https://codesandbox.io/p/sandbox/infallible-haibt-e3lcd4

File: examples/helia-vite-react/src/hooks/useCommitText.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          })
+        }
+        setCommitedText(text)
+      } catch (e) {
+        console.error(e)
+      }
+    } else {
+      console.log('please wait for helia to start')
+    }
+  }, [error, starting, cid, helia]) // make sure to add helia in the dependencies array of the fs will suspend
```
[the remark this answers is not in the document]
Comment (polus-arcticus) [reply examples/helia-vite-react/src/hooks/useCommitText.jsx]: https://github.com/polus-arcticus/helia-examples/blob/feature/helia-react-vite/examples/helia-vite-react/public/Peek%202023-03-22%2013-21.gif
is a gif illustrating what I mean, I added it as i'm not sure if this behaviour from unixfs is expected

[the remark this answers is not in the document]
Comment (polus-arcticus) [reply examples/helia-vite-react/src/hooks/useCommitText.jsx]: https://github.com/polus-arcticus/helia-examples/blob/38c9bc955b211e755fa0242c1b7a83691d502608/examples/helia-vite-react/src/hooks/useCommitText.jsx#L48

**Comment by polus-arcticus** [comment]
> LGTM, but the comment about helia dep should be enough. Can you remove the extra gifs?

you bet, i probably just could've uploaded them to imgur or something

**Review by SgtPooki — APPROVED** [review]

**Comment by SgtPooki** [comment]
@polus-arcticus thanks so much for the contribution!! If you want to help out more, there's a group of prioritized examples at https://github.com/ipfs/helia/issues/43! just ping me on that issue if you want to grab one and I can put your name next to it so we can make sure any other contributors aren't stepping on toes.

**Comment by polus-arcticus** [comment]
> @polus-arcticus thanks so much for the contribution!! If you want to help out more, there's a group of prioritized examples at [ipfs/helia#43](https://github.com/ipfs/helia/issues/43)! just ping me on that issue if you want to grab one and I can put your name next to it so we can make sure any other contributors aren't stepping on toes.

Thanks @SgtPooki ! okay sounds good :+1:

**Pull request merged**